xbmc/semantic/search/API_REFERENCE.md

```diff
@@ -0,0 +1,518 @@
+
+**Parameters:**
+- `mediaId` - Media item ID
+- `mediaType` - Media type
+- `timestampMs` - Center timestamp in milliseconds
+- `windowMs` - Window size in milliseconds (default: 60000 = ±60s)
+
+**Returns:**
+- Vector of `SemanticChunk` objects within the time window
+- Ordered by start timestamp
+- Empty vector on error
+
+**Example:**
+```cpp
+// Get context around 5-minute mark
+int64_t timestamp = 300000;  // 5 minutes
+int64_t window = 30000;      // ±30 seconds
+
+auto context = search.GetContext(episodeId, "episode", timestamp, window);
+
+// Returns chunks from 4:30 to 5:30
+for (const auto& chunk : context)
+{
+    CLog::Log(LOGINFO, "{}-{}ms: {}",
+              chunk.startMs, chunk.endMs, chunk.text);
+}
+```
+
+---
+
+#### `std::vector<SemanticChunk> GetMediaChunks(int mediaId, const std::string& mediaType)`
+
+Get all indexed chunks for a media item.
+
+**Parameters:**
+- `mediaId` - Media item ID
+- `mediaType` - Media type
+
+**Returns:**
+- Vector of all `SemanticChunk` objects for the media
+- Ordered by start timestamp
+- Empty vector on error or no chunks
+
+**Example:**
+```cpp
+auto chunks = search.GetMediaChunks(movieId, "movie");
+
+CLog::Log(LOGINFO, "Movie has {} chunks", chunks.size());
+
+// Calculate total indexed duration
+if (!chunks.empty())
+{
+    int64_t totalMs = chunks.back().endMs - chunks.front().startMs;
+    CLog::Log(LOGINFO, "Indexed duration: {} minutes", totalMs / 60000);
+}
+```
+
+---
+
+### Status and Statistics
+
+#### `bool IsMediaSearchable(int mediaId, const std::string& mediaType)`
+
+Check if a media item is indexed and searchable.
+
+**Parameters:**
+- `mediaId` - Media item ID
+- `mediaType` - Media type
+
+**Returns:**
+- `true` if media has completed indexing and has chunks
+- `false` if not indexed, in progress, or failed
+
+**Criteria for Searchable:**
+- At least one source (subtitle/transcription/metadata) has status COMPLETED
+- Chunk count > 0
+
+**Example:**
+```cpp
+if (search.IsMediaSearchable(movieId, "movie"))
+{
+    // Safe to search
+    auto results = search.SearchInMedia("explosion", movieId, "movie");
+}
+else
+{
+    // Show "Indexing in progress" message
+    ShowIndexingDialog(movieId);
+}
+```
+
+---
+
+#### `IndexStats GetSearchStats()`
+
+Get overall search database statistics.
+
+**Returns:**
+- `IndexStats` structure with counts and metrics
+- Zero values on error
+
+**IndexStats Structure:**
+```cpp
+struct IndexStats {
+    int totalMedia;       // Total media items in system
+    int indexedMedia;     // Media with completed indexing
+    int totalChunks;      // Total number of chunks
+    int totalWords;       // Approximate word count
+    int queuedJobs;       // Pending indexing jobs
+};
+```
+
+**Example:**
+```cpp
+IndexStats stats = search.GetSearchStats();
+
+float indexingProgress =
+    (float)stats.indexedMedia / stats.totalMedia * 100.0f;
+
+CLog::Log(LOGINFO, "Indexing progress: {:.1f}%", indexingProgress);
```

Comment:
Please update to the current log standards using `{}`.

For more information please see the [libfmt documentation](https://fmt.dev/latest/syntax.html)